src/extension.ts

```diff
@@ -1,40 +1,58 @@
 'use strict';
 
 import * as vscode from 'vscode';
+import * as path from 'path';
 
-const fetchPackageName = async (context: vscode.ExtensionContext) => {
-    const files: vscode.Uri[] = await vscode.workspace.findFiles('pubspec.yaml');
-    if (files.length !== 1) {
-        vscode.window.showErrorMessage(`Expected to find a single pubspec.yaml file, ${files.length} found.`);
+const fetchPackageInfoFor = async (activeDocumentUri: vscode.Uri) => {
+    const pubspecUris = await findPubspec(activeDocumentUri);
+    if (pubspecUris.length != 1) {
+        vscode.window.showErrorMessage(`Expected to find a single pubspec.yaml file above ${activeDocumentUri}, ${pubspecUris.length} found.`);
         return null;
     }
-    const file : vscode.TextDocument = await vscode.workspace.openTextDocument(files[0]);
-    const fileName = file.fileName.replace(/\/pubspec\.yaml$/, '');
-    const possibleNameLines = file.getText().split('\n').filter((line: String) => line.match(/^\s*name:/));
+
+    const pubspec : vscode.TextDocument = await vscode.workspace.openTextDocument(pubspecUris[0]);
+    const projectRoot = path.dirname(pubspec.fileName);
+    const possibleNameLines = pubspec.getText().split('\n').filter((line: String) => line.match(/^\s*name:/));
     if (possibleNameLines.length !== 1) {
         vscode.window.showErrorMessage(`Expected to find a single line starting with 'name:' on pubspec.yaml file, ${possibleNameLines.length} found.`);
         return null;
     }
     const nameLine = possibleNameLines[0];
-    const regex = /^\s*name:\s*(.*)$/mg.exec(nameLine);
-    if (!regex) {
+    const packageNameMatch = /^\s*name:\s*(.*)$/mg.exec(nameLine);
```

Comment:
I think the `^\s*` at the start of this regex might be too lenient.

Reply:
Do you think it's possible the pubspec has multiple name: entries? That has never occurred to me... I'll approve this one, if you find a better way to perform this match, please open another PR :)

Reply:
Yep. The pubspec is pretty free-form in terms of what you can put in it. You might have noticed that flutter adds their own section. There's no reason other projects couldn't do the same.

`name` seems like a common enough key, and the `name` we care about always sits top level. So enforcing it is a good idea.

Reply:
@shyndman makes sense, please open another issue for this potential problem, and if you find an elegant solution, you can open another PR :) I'm not sure simply counting tabs is totally fail safe either, maybe you need to compare with the first line indentation? not sure, we gotta think this through